### PR TITLE
Feature/faas 1151 stabilise vsf deps fixed

### PR DESCRIPTION
**Pull request by abogdanov-work**
## PR Description
This PR upgrades VSF theme dependencies to a ~ versions and resolves all related issues

## Checklist
- [x] I agree with the Code Contribution License Agreement in CONTRIBUTING.md

### CODE REVIEW

File: theme/nuxt.config.js [hunk 1]
```diff
@@ -1,11 +1,6 @@
 import webpack from 'webpack';
 
 export default {
-  mode: 'universal',
```
Comment (abogdanov-work) [inline theme/nuxt.config.js:4]: This was removed since it's shouldn't be specified here, it's default configs and specifying them here will block normal Nuxt behavior in regards to customizations via .env file

File: theme/nuxt.config.js [hunk 2]
```diff
@@ -177,14 +172,41 @@ export default {
     },
   },
   i18n: {
-    defaultLocale: process.env.LOCALE_DEFAULT || 'en_US',
```
Comment (abogdanov-work) [inline theme/nuxt.config.js:180]: i18n configs were updated to match new nuxt-i18n-localization plugin that was introduced in @vue-storefront/nuxt 2.3.4

File: package.json [hunk 3]
```diff
@@ -11,33 +11,26 @@
...
-    "@vue-storefront/nuxt": "2.3.1",
-    "@vue-storefront/nuxt-theme": "2.3.1",
-    "cookie-universal-nuxt": "2.1.4",
-    "core-js": "2.6.12",
-    "dotenv": "10.0.0",
-    "nuxt": "2.14.12",
-    "nuxt-i18n": "6.20.0",
-    "vee-validate": "3.4.5",
-    "vue-scrollto": "2.20.0"
+    "@storefront-ui/vue": "~0.10.5",
```
Comment (abogdanov-work) [inline package.json:20]: 0.10.5 version is the latest we can use without breaking changes, this version introduces several build warnings in the internal VSF-UI components which aren't affecting the build and were resolved in later version of the lib

File: package.json [hunk 4]
```diff
@@ -11,33 +11,26 @@
...
+    "@vue-storefront/middleware": "~2.3.4",
```
Comment (abogdanov-work) [inline package.json:21]: 2.3.4 version is the latest we can use without updates to spryker-vsf packages, such as composables and api-client

File: package.json [hunk 5]
```diff
@@ -11,33 +11,26 @@
...
+    "@vue-storefront/nuxt": "~2.3.4",
```
Comment (abogdanov-work) [inline package.json:22]: 2.3.4 version is the latest we can use without updates to spryker-vsf packages, such as composables and api-client.
Note: new nuxti18nlocalization plugin was introduced in this version which requires some additional properties of the i18n configuration object

File: package.json [hunk 6]
```diff
@@ -11,33 +11,26 @@
...
+    "@vue-storefront/nuxt-theme": "~2.3.4",
```
Comment (abogdanov-work) [inline package.json:23]: 2.3.4 version is the latest we can use without updates to spryker-vsf packages, such as composables and api-client

File: package.json [hunk 7]
```diff
@@ -11,33 +11,26 @@
...
+    "cookie-universal-nuxt": "^2.1.5",
+    "core-js": "^2.6.12",
```
Comment (abogdanov-work) [inline package.json:25]: All versions with caret were updated to the latest available version, accept **core-js** which was updated to the latest v.2 version to avoid breaking changes

File: package.json [hunk 8]
```diff
@@ -11,33 +11,26 @@
...
+    "dotenv": "^10.0.0",
+    "nuxt": "~2.14.12",
```
Comment (abogdanov-work) [inline package.json:27]: version of Nuxt higher than 2.14.12 can't be used due to issues inside "@vue/composition-api": "1.0.0-beta.21" which is required by all VSF core dependencies

File: package.json [hunk 9]
```diff
@@ -11,33 +11,26 @@
...
+    "nuxt-i18n": "^6.28.1",
+    "vee-validate": "^3.4.14",
+    "vue-scrollto": "^2.20.0"
   },
   "devDependencies": {
-    "@nuxt/types": "0.7.9",
-    "@nuxt/typescript-build": "2.0.4",
-    "@vue/test-utils": "1.1.3",
-    "babel-jest": "24.9.0",
```
Comment (abogdanov-work) [inline package.json:36]: testing related dependencies were removed until we will properly define e2e process

File: package.json [hunk 10]
```diff
@@ -11,33 +11,26 @@
...
-    "jest": "24.9.0",
-    "vue-jest": "4.0.0-rc.1"
-  },
-  "resolutions": {
```
Comment (abogdanov-work) [inline package.json:40]: resolutions no longer needed with the current set of dependencies, v. 7.12.13 is selected automatically

File: theme/nuxt.config.js [hunk 11]
```diff
@@ -177,14 +172,41 @@ export default {
...
+      { name: 'NL', label: 'Netherlands' }
+    ],
+    currencies: [
+      { name: 'EUR', label: 'Euro' },
+      { name: 'USD', label: 'Dollar' }
+    ],
     locales: [
-      { code: 'en_US', iso: 'en_US', file: 'en.js', label: 'English' },
-      { code: 'de_DE', iso: 'de_DE', file: 'de.js', label: 'German' },
+      { code: 'en', label: 'English', file: 'en.js', iso: 'en' },
```
Comment (dunqan) [inline theme/nuxt.config.js:188]: Is this change really needed? We want to support 'en_US' as a locale.
